src-tool/Pact/Analyze/Eval/Core.hs

```diff
@@ -231,6 +248,41 @@ evalCore (StrContains needle haystack) = do
     _sSbv (coerceS @Str @String needle')
     `SBVS.isInfixOf`
     _sSbv (coerceS @Str @String haystack')
+
+-- hash values
+-- TODO (RS): we should add warnings to the stack, allowing
+--            to return shims in case, the content is not statically known
```

Comment:
This TODO seems to be stale

Reply:
Actually this TODO is still relevant. Right now, we do only support hashes (and a lot of other natives) for statically known data. We have no notion of warnings at this stage and can only fail. In the future, I would like to emit a warning at this stage and shim a value.